core/aws_ddk_core/stages/appflow.py

```diff
@@ -0,0 +1,224 @@
+        source_flow_config: Optional[CfnFlow.SourceFlowConfigProperty] = None,
+        tasks: Optional[List[CfnFlow.TaskProperty]] = None,
+        state_machine_input: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        DDK AppFlow stage.
+
+        Stage that contains a step function that runs an AppFlow flow.
+        The flow must be pre-created.
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the stage
+        environment_id : str
+            Identifier of the environment
+        flow_name : Optional[str]
+            Name of the AppFlow flow to run
+        destination_flow_config : Optional[CfnFlow.DestinationFlowConfigProperty]
+            The flow destination_flow_config properties
+        source_flow_config : Optional[CfnFlow.SourceFlowConfigProperty]
+            The flow source_flow_config properties
+        tasks : Optional[List[CfnFlow.TaskProperty]]
+            The flow tasks properties
+        state_machine_input : Optional[Dict[str, Any]]
+            Input of the state machine
+        """
+        super().__init__(scope, id)
+
+        self._state_machine_input: Optional[Dict[str, Any]] = state_machine_input
+        self._event_detail_type: str = f"{id}-event-type"
+
+        # If AppFlow flow name is not supplied, create one
+        self._flow: Optional[CfnFlow] = None
+        if not flow_name:
+            self._flow = CfnFlow(
+                self,
+                f"{id}-flow",
+                destination_flow_config_list=[destination_flow_config],
+                source_flow_config=source_flow_config,
+                tasks=tasks,
+                trigger_config=CfnFlow.TriggerConfigProperty(trigger_type="OnDemand"),
+            )
+            flow_name = self._flow.name
+
+        # Create lambda used to check flow execution status
+        status_lambda_role = Role(
+            self,
+            "flow-execution-status-lambda-role",
+            assumed_by=ServicePrincipal("lambda.amazonaws.com"),
+            description="lambda role to check appflow flow execution status",
+        )
+
+        status_lambda = LambdaFactory.function(
+            self,
+            id="flow-execution-status-lambda",
+            environment_id=environment_id,
+            code=Code.from_asset(str(Path(__file__).parent.joinpath("appflow_lambda/"))),
+            handler="index.lambda_handler",
+            role=status_lambda_role,
+        )
+        # Enable the function to get flow execution records
+        status_lambda.add_to_role_policy(
+            PolicyStatement(
+                effect=Effect.ALLOW,
+                actions=[
+                    "appflow:DescribeFlowExecutionRecords",
+                ],
+                resources=["*"],
+            )
+        )
+
+        # Create start flow step function task
+        flow_object = CustomState(
```

Comment:
One idea is to make the constructor shorter and more readable perhaps we should move some code into private methods e.g. `_create_start_flow_custom_task(flow_name: str, ...)`, etc.

Reply:
Do you need the variables like `flow_object` in `_create_start_flow_custom_task`? I think you can just `return CustomState(...)`